File: benchmarks/poly/factorial/factorial.py
import sys, time


def fact(n):
    if n < 2:
        return 1
    return n * fact(n - 1)


def main():
    _t0 = time.perf_counter_ns()
    for i in range(10):
        print(fact(i))
    print(f"bench_ns={time.perf_counter_ns() - _t0}", file=sys.stderr)


if __name__ == "__main__":
    main()

File: benchmarks/poly/factorial/factorial.go
package main

import (
	"fmt"
	"os"
	"time"
)

func fact(n int) int {
	if n < 2 {
		return 1
	}
	return n * fact(n-1)
}

func main() {
	t0 := time.Now()
	for i := 0; i < 10; i++ {
		fmt.Println(fact(i))
	}
	fmt.Fprintf(os.Stderr, "bench_ns=%d\n", time.Since(t0).Nanoseconds())
}
